india_compliance/gst_india/utils/gstr/gstr.py

```diff
@@ -122,3 +123,64 @@ def set_key(self, key, value):
 
     def update_gstins(self):
         pass
+
+
+@frappe.whitelist()
+def validate_company_gstins(company=None, company_gstin=None):
+    """
+    Checks the validity of the company's GSTIN authentication.
+
+    Args:
+        company_gstin (str): The GSTIN of the company to validate.
+
+    Returns:
+        dict: A dictionary where the keys are the GSTINs and the values are booleans indicating whether the authentication is valid.
+    """
+    frappe.has_permission("GST Settings", throw=True)
```

Comment:
Here you are checking for read permissions and below for write permissions. How about checking for write permissions at all places, considering the end objective is to update GST Credentials with authtokens.